### ISSUE TITLE
testsuite failing on 32bit arches

### ISSUE THREAD

**Issue by juhp** [fosskers/microlens-aeson#5]
I haven't investigated but today I discovered that microlens-aeson-2.3.0.4 tests are failing Fedora i686 and armv7hl like this:

```
:
  Non-primitive Traversals
    : OK
    : OK
    : OK
    : OK
    : FAIL
      test/Test.hs:51:
      expected: [Number 4.0,Number 7.0]
       but got: [Number 7.0,Number 4.0]
    : FAIL
      test/Test.hs:52:
      expected: "{\"a\":40,\"b\":70}"
       but got: "{\"b\":70,\"a\":40}"
    : OK
    : OK
    : OK
    : OK
    : OK
2 out of 33 tests failed (0.00s)
```

See for example: https://koji.fedoraproject.org/koji/taskinfo?taskID=39485695

**Comment by fosskers** [comment]
Looks like the same issue as #1 , and `lens-aeson` seems to have this fixed: https://github.com/lens/lens-aeson/issues/30

**Comment by iliastsi** [comment]
This affects Debian as well. I fixed it by applying [this](https://salsa.debian.org/haskell-team/DHG_packages/-/blob/40b5c9f76c77b883c5dfd27876e0c5c0995090d4/p/haskell-microlens-aeson/debian/patches/fix-tests) patch. Please let me know if you want me to provide a PR.

**Comment by fosskers** [comment]
Yes please!

**Comment by fosskers** [comment]
Released as `2.3.1`.
